src/hotspot/share/runtime/sharedRuntime.cpp

```diff
@@ -2684,16 +2684,20 @@ extern "C" void unexpected_adapter_call() {
 }
 
 static void post_adapter_creation(const AdapterBlob* new_adapter, const AdapterHandlerEntry* entry) {
-  char blob_id[256];
-  jio_snprintf(blob_id,
-                sizeof(blob_id),
-                "%s(%s)",
-                new_adapter->name(),
-                entry->fingerprint()->as_string());
-  Forte::register_stub(blob_id, new_adapter->content_begin(), new_adapter->content_end());
-
-  if (JvmtiExport::should_post_dynamic_code_generated()) {
-    JvmtiExport::post_dynamic_code_generated(blob_id, new_adapter->content_begin(), new_adapter->content_end());
+  if (Forte::is_enabled() || JvmtiExport::should_post_dynamic_code_generated()) {
+    char blob_id[256];
+    jio_snprintf(blob_id,
+                 sizeof(blob_id),
+                 "%s(%s)",
+                 new_adapter->name(),
+                 entry->fingerprint()->as_string());
+    if (Forte::is_enabled()) {
+      Forte::register_stub(blob_id, new_adapter->content_begin(), new_adapter->content_end());
+    }
+
+    if (JvmtiExport::should_post_dynamic_code_generated()) {
+      JvmtiExport::post_dynamic_code_generated(blob_id, new_adapter->content_begin(), new_adapter->content_end());
+    }
```

Comment:
The lines 2698-2670 is better to move out of the if-statement at the line 2687.

Reply:
HI Serguei, thanks for the review. Lines 2698-2700 need the blob_id which is generated at line 2688, so they have to stay inside the outer "if" block.

Reply:
Thanks, Ioi. You are right.